R/widerRepeated.R

```diff
@@ -0,0 +1,123 @@
+#' @name widerRepeated
+#' @title Transform Data Into Wide Format
+#' 
+#' @description Converts a dataframe into wide format given the column names to expand.
+#'   
+#' @param Records A \code{data.frame} containing the records from
+#'        \code{\link{exportRecordsTyped}}
+#' @param rcon A REDCap connection object as created by \code{unlockREDCap}.
+#' @param idvar \code{character}, Names of one or more variables in long format that identify multiple records belonging to the same group. 
+#'        These variables may also be present in wide format. (e.g., 'record_id')
+
+
+widerRepeated <- function(rcon, idvar){
+  # Records <- reshape(Records, idvar = idvar, timevar = timevar, direction = 'wide')
+  # export data dictionary
+  meta <- exportMetaData(rcon)
+  
+  # export repeating
+  repeating <- exportRepeatingInstrumentsEvents(rcon)
+  
+  # export events
+  events <- exportEvents(rcon)
+  
+  # export mappings
+  mappings <- exportMappings(rcon)
+  
+  # export data
+  records <- exportRecordsTyped(rcon)
+  
+  # helper function
+  is.blank <- function(x) is.na(x) | x == ""
+  
+  # set everything as character in data and data dictionary and assign id.temp to data
+  dat <- data.frame(lapply(records, as.character), stringsAsFactors = FALSE)
+  names(dat)[names(dat) == idvar] <- "id.tmp"
+  
+  dd <- data.frame(lapply(meta, as.character), stringsAsFactors = FALSE)
+  
+  # get correct event mappings if redcap_event_name is present or not
+  if ("redcap_event_name" %in% names(dat)) {
+    #get the event-form mapping
+    event.map <- mappings
+  } else {
+    event.map = data.frame(
+      unique_event_name = "",
+      form = sort(unique(dd$form_name)))
+  }
+  
+  # because not all projects will have all id modifier fields
+  df.all_ids <- data.frame(id.tmp = character(0), redcap_event_name = character(0), 
+                           redcap_repeat_instrument = character(0), redcap_repeat_instance = character(0))
+  tmpd <- merge(
+    data.frame(dat),
+    df.all_ids,
+    by = names(df.all_ids)[
+      names(df.all_ids) %in% names(dat)],
+    all = TRUE)
+  tmpd[is.na(tmpd)] <- ""
+  
+  id.fields = names(df.all_ids)
+  
+  map <- event.map
+  data.frame(map)
+  map[is.na(map)] <- ""
+  
+  for (i in sort(unique(dd$form_name))) {
+    # print(i)
+    vars.tmp <- names(dat)[sub("___.*", "", names(dat)) %in% dd$field_name[dd$form_name == i]]
```

Comment:
This is _really_ obscure...using the following would be more consistent with how we handle checkbox field names elsewhere in the package

`sub(REGEX_CHECKBOX_FIELD_NAME, "\\1", names(dat))`